Tests/NIOPosixTests/RawSocketBootstrapTests.swift

```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import NIOCore
+import NIOPosix
+
+#if canImport(Darwin)
+import Darwin
+#endif
+
+extension NIOIPProtocol {
+    static let reservedForTesting = Self(rawValue: 253)
+}
+
+final class RawSocketBootstrapTests: XCTestCase {
+    func testWriteAndRead() throws {
+        #if canImport(Darwin)
+        try XCTSkipIf(geteuid() != 0, "Raw Socket API requires root privileges on Darwin")
+        #endif
+        
+        let elg = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { XCTAssertNoThrow(try elg.syncShutdownGracefully()) }
+        let channel = try RawSocketBootstrap(group: elg)
```

Comment:
We probably need a test of the `ip_hdrincl` path.

Reply:
Tested now in `RawSocketBootstrapTests.testIpHdrincl`